### DISCUSSION TITLE
Making an existing field have a relationship

### DISCUSSION THREAD

**Discussion by OfficialOzzy** [directus/directus#3415] [Feature Requests]
It seems that any existing fields with foreign keys do not allow you to setup a relationship, only when creating new fields.

It would be great if we could add this feature as most databases are existing and not brand new.

**Comment by rijkvanzanten** [comment]
> as most databases are existing and not brand new.

Any analytics to back this up? While existing DBs is definitely a use case we support, I can't conclusively say that _most_ Directus installations are on existing DBs 🙂

**Reply by WoLfulus** [reply]
I'd rephrase what @mrbenosborne said to something like "as there are schemas that are created outside of Directus" because I faced this myself while having one being managed/created/migrated by something else other than Directus. So I'd love that too! 👍 

I'm not sure, but I think I even mentioned this to you a while ago.

**Reply by OfficialOzzy** [reply]
> > as most databases are existing and not brand new.
> 
> Any analytics to back this up? While existing DBs is definitely a use case we support, I can't conclusively say that _most_ Directus installations are on existing DBs 🙂

Very true indeed! However new projects might not use Directus to manage their database and prefer something like a framework such as Laravel to manage migrations etc

It would be interesting to find out some stats on existing projects vs new and if the user were to start a fresh project would they start their database planning in Directus.

**Reply by OfficialOzzy** [reply]
Also I work for a travel company and we are looking at using Directus but this is the only thing holding us back from using it as we have an existing database already.

**Reply by benhaynes** [reply]
A big goal/purpose of DIrectus is to mirror the existing database schema/data... so this is something that we're definitely focused on. We're super careful about collecting too much user info, but it _would_ be really nice to have more insight into if new or existing DBs are used. Though that's hard to capture, since you might either install on top of an existing DB, or import it after install. Still, this is something we'll 100% support... just busy times over here at the moment. :)

**Reply by OfficialOzzy** [reply]
Hey Ben,

Yeah no worries at all, I've been following the release and you guys never seem to sleep 😂

Good work all round, I'd be more than happy to help if there's any tickets that can be picked up.

**Reply by apstanisic** [reply]
Also all migrations from v8 to v9 are essentially existing databases.

**Comment by LazyBun** [comment]
Hello. 

Is it possible to handle existing DB relationships in Directus or is it still a work in progress? :)

**Reply by rijkvanzanten** [reply]
What do you mean by _handle existing DB relationships_? Directus currently uses foreign key constraints to determine what fields are considered relational, so if your database has relationships with foreign key constraints configured, everything should work as expected out of the box 🙂

**Reply by jvmejias** [reply]
I have a similar problem. I'm starting to manage an existing database in Directus 9.6 and I'm unable to set a M2O relationship. The relationship tag is simply missing! I spent the whole day with this and I'm stuck. I guess there is something in my database (not mine actually) that Directus doesn't like, be it charset, FKs, column names... I don't know. I have set M2O relationships in previous Directus 6, 7 and 8, but it's not working "out of the box" for me in this version.
Both primary and foreign key are type INT(11) NOT NULL AUTO_INCREMENT
![1](https://user-images.githubusercontent.com/28959407/159783542-802261fd-52df-4beb-a56a-b481627535a1.jpg)
.

**Reply by rijkvanzanten** [reply]
> Both primary and foreign key are type INT(11) NOT NULL AUTO_INCREMENT

Do your foreign keys have foreign key constraints to the key they're referencing? That's what Directus introspects for the relational fields 👍🏻

**Reply by kevlarr** [reply]
This seems fine for 1:many relationships but how would one get Directus to pick up an existing intersection table (or, "junction" table as you guys call them) for a many:many relationship? I can verify it's named accordingly and that foreign keys are set up properly, since if I manually create a many:many relationship field through Directus UI then it winds up creating almost the exact table I do, with the only difference being allowing the foreign key columns to be nullable and using a sequence instead of identity for the primary key.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by adamkhan** [comment]
I have the same problem in v10.1, and I imagine it's pretty common. I didn't understand @rijkvanzanten's explanation that foreign keys need foreign key restraints — can you explain further? What should we be doing to an Integer field to have the possibility of setting it up as a relationship field?

What I've done as a workaround when migrating an existing db to Directus is create another field within Directus, setting up the relationship there. Then I make direct changes to the directus_fields and directus_relations tables. In directus_fields I set the value of my field's "special" column to "m2o", and in the entry newly created in directus_relations for my new directus field, I change the "many_field" value to the original field name. Then I delete the new field.

**Comment by tonnyavery** [comment]
I know this topic was discussed long time ago, but I would love to bring it up again.
I have a quite big and old database that doesn't have correct relationships and foreign keys configured.
It would be lovely if I could do that in Directus while configuring my collections.
